File: src/leetcode/python/PowerOfThree_326.py
class PowerOfThree_326(object):
    def isPowerOfThree(self, n):
        """
        :type n: int
        :rtype: bool
        """
        if n <= 0 and (n & 1) == 0:
            return False
        return n == 1 or n % 3 == 0 and self.isPowerOfThree(n / 3)

File: src/leetcode/java/PowerOfThree_326.java
package leetcode.java;

public class PowerOfThree_326 {
  public boolean isPowerOfThree(int n) {
    if (n <= 0 || (n & 1) == 0) return false;
    return (n == 1 || n % 3 == 0 && isPowerOfThree(n / 3));
  }
}
